rewrite-yaml/src/test/java/org/openrewrite/yaml/HelmTemplateParsingTest.java

```diff
@@ -0,0 +1,323 @@
+                    memory: {{ .Values.resources.limits.memory | default "512Mi" }}
+                    cpu: {{ .Values.resources.limits.cpu | default "500m" }}
+              """
+          )
+        );
+    }
+
+    @Test
+    void parseHelmTemplateInFlowStyle() {
+        rewriteRun(
+          yaml(
+            """
+              metadata:
+                labels: { app: {{ .Values.appName }}, version: {{ .Values.version }} }
+                annotations: {
+                  "helm.sh/hook": post-install,
+                  "helm.sh/hook-weight": "{{ .Values.hookWeight }}"
+                }
+              data:
+                config.json: '{"apiUrl": "{{ .Values.apiUrl }}", "debug": {{ .Values.debug }}}'
+              """
+          )
+        );
+    }
+
+    @ExpectedToFail("Invalid YAML syntax with Helm template expressions")
+    @Test
+    void parseHelmTemplateWithToYaml() {
+        rewriteRun(
+          yaml(
+            """
+              spec:
+                {{- with .Values.nodeSelector }}
+                nodeSelector:
+                  {{- toYaml . | nindent 8 }}
+                {{- end }}
+                {{- with .Values.affinity }}
+                affinity:
+                  {{- toYaml . | nindent 8 }}
+                {{- end }}
+                {{- with .Values.tolerations }}
+                tolerations:
+                  {{- toYaml . | nindent 8 }}
+                {{- end }}
+              """
+          )
+        );
+    }
+
+    @Test
+    void preserveFormattingWithHelmTemplates() {
+        rewriteRun(
+          yaml(
+            """
+              metadata:
+                name:     {{ .Release.Name }}
+                namespace:    {{ .Values.namespace }}
+                labels:
+                  app:    {{ .Values.appName }}
+                  chart:    {{ .Chart.Name }}-{{ .Chart.Version }}
+              spec:
+                replicas:    {{ .Values.replicas }}
+              """
+          )
+        );
+    }
+
+    @Test
+    void helmTemplatesWithComments() {
+        rewriteRun(
+          yaml(
+            """
+              # This is a Helm chart
+              apiVersion: v1
+              kind: Service
+              metadata:
+                name: {{ .Release.Name }} # The release name
+                labels:
+                  # Standard labels
+                  app: {{ .Values.appName }}
+                  # Chart information
+                  chart: {{ .Chart.Name }}
+              spec:
+                # Service configuration
+                type: {{ .Values.service.type | default "ClusterIP" }} # Default to ClusterIP
+                ports:
+                  - port: {{ .Values.service.port }} # The service port
+              """
+          )
+        );
+    }
+
+    @Test
+    void multilineHelmTemplates() {
+        rewriteRun(
+          yaml(
+            """
+              data:
+                script.sh: |
+                  {{- if .Values.enableDebug }}
```

Comment:
Conditional can either use `{{-` or `{{ `. May be worth adding a test case for.

https://helm.sh/docs/chart_template_guide/control_structures/#ifelse